bugrap/src/main/java/org/vaadin/bugrap/ui/ReportDetailView.java

```diff
@@ -0,0 +1,173 @@
+package org.vaadin.bugrap.ui;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.List;
+
+import org.vaadin.bugrap.BaseModel;
+import org.vaadin.bugrap.domain.entities.Comment;
+import org.vaadin.bugrap.domain.entities.Report;
+import org.vaadin.bugrap.ui.generated.ReportDetailViewBase;
+
+import com.vaadin.data.Binder;
+import com.vaadin.server.FileResource;
+import com.vaadin.ui.HorizontalLayout;
+import com.vaadin.ui.Link;
+import com.vaadin.ui.Notification;
+import com.vaadin.ui.Notification.Type;
+import com.vaadin.ui.ProgressBar;
+import com.vaadin.ui.Upload.ProgressListener;
+import com.vaadin.ui.Upload.Receiver;
+import com.vaadin.ui.Upload.StartedEvent;
+import com.vaadin.ui.Upload.StartedListener;
+import com.vaadin.ui.Upload.SucceededEvent;
+import com.vaadin.ui.Upload.SucceededListener;
+import com.vaadin.ui.VerticalLayout;
+import com.vaadin.ui.Window;
+
+public class ReportDetailView extends ReportDetailViewBase implements Receiver, SucceededListener, StartedListener, ProgressListener{
+
+
+	private ReportsModel model;
+	private Binder<Report> binder = new Binder<Report>();
+	
+	public ReportDetailView(ReportsModel reportModel) {
+		model = reportModel;
+		initializeBinder();
+		initializeUIComponents();
+		init();
+	}
+	
+
+	private void initializeBinder() {
+		binder.bind(cmbEditPriority, Report::getPriority, Report::setPriority);
+		binder.bind(cmbEditType, Report::getType, Report::setType);
+		binder.bind(cmbEditStatus, Report::getStatus, Report::setStatus);
+		binder.bind(cmbEditAssignedTo, Report::getAssigned, Report::setAssigned);
+		binder.bind(cmbEditVersion, Report::getVersion, Report::setVersion);
+	}
+
+	private void initializeUIComponents() {
+		btnUpdateReport.addClickListener(evt -> saveReport());
+		btnRevertReport.addClickListener(evt -> revertChanges());
+		btnDone.addClickListener(evt -> saveComment());
+		btnCancel.addClickListener(evt -> closeWindow());
+		
+		btnUpload.setReceiver(this);
+		btnUpload.addStartedListener(this);
+		btnUpload.addSucceededListener(this);
+		btnUpload.addProgressListener(this);
+	}
+
+	private void init() {
+		((HorizontalLayout)pnlAttachments.getContent()).removeAllComponents();
+		pnlAttachments.setVisible(false);
+		Report report = model.getReportForEdit();
+		binder.setBean(model.getReportForEdit());
+		lblProjectSummary.setValue(report.getSummary());
+		initializeComboContents();
+		fillComments();
+	}
+	
+	private void initializeComboContents() {
+		cmbEditPriority.setItems(model.getPriorties());
+		cmbEditType.setItems(model.getTypes());
+		cmbEditStatus.setItems(model.getStatuses());
+		cmbEditAssignedTo.setItems(model.findReporters());
+		cmbEditVersion.setItems(model.findProjectVersions(model.getReportForEdit().getProject()));
```

Comment:
The model is already aware of the project.

Reply:
changed